### PR TITLE
Mark classes and interfaces as API part 6

### LINKED ISSUE #32675: Mark classes and interfaces as API part 6
Mark the following classes and interfaces as API:

1. Magento\Checkout\Model\Type\Onepage
1. Magento\Framework\Image\AdapterFactory
1. Magento\Catalog\Model\Config
1. Magento\Customer\Model\CustomerRegistry
1. Magento\Quote\Setup\QuoteSetup
1. Magento\Ui\Component\Control\Container
1. Magento\Catalog\Model\Layer\Filter\Item

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Marked the following classes and interfaces as API

Magento\Checkout\Model\Type\Onepage
Magento\Framework\Image\AdapterFactory
Magento\Catalog\Model\Config
Magento\Customer\Model\CustomerRegistry
Magento\Quote\Setup\QuoteSetup
Magento\Ui\Component\Control\Container
Magento\Catalog\Model\Layer\Filter\Item

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32675

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Layer/Filter/Item.php [hunk 1]
```diff
@@ -3,14 +3,15 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
+namespace Magento\Catalog\Model\Layer\Filter;
 
 /**
- * Filter item model
+ * Class \Magento\Catalog\Model\Layer\Filter\Item
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/Layer/Filter/Item.php]: Please remove this line, it's absolutely useless

File: app/code/Magento/Checkout/Model/Type/Onepage.php [hunk 2]
```diff
@@ -15,6 +17,9 @@
 use Magento\Sales\Model\Order\Email\Sender\OrderSender;
 
 /**
+ * Class \Magento\Checkout\Model\Type\Onepage
```
Comment (ihor-sviziev) [inline app/code/Magento/Checkout/Model/Type/Onepage.php]: Please remove this line, it's absolutely useless

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please fix failing unit tests

**Comment by mrtuvn** [comment]
Seem we have issue in this line after add strict_type
https://github.com/magento/magento2/blob/3f89f034f1b9792f66b3364623ee4a512a286e58/app/code/Magento/Catalog/Model/Config.php#L403

**Comment by ihor-sviziev** [comment]
@mrtuvn I believe it's better to revert adding strict  type for this file if it's introducing some issues

File: app/code/Magento/Catalog/Model/Config.php [hunk 3]
```diff
@@ -3,6 +3,7 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/Config.php]:
```suggestion
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Model/Config.php (hunk 3 above)
Comment (mrtuvn) [reply app/code/Magento/Catalog/Model/Config.php]: i have reverted that too

**Comment by mrtuvn** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8994](https://jira.corp.magento.com/browse/ENGCOM-8994) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE

**Comment by gabrieldagama** [comment]
@magento import pr to magento-engcom/magento2ce

**Comment by gabrieldagama** [comment]
@magento import pr to https://github.com/magento-engcom/magento2ce

**Comment by gabrieldagama** [comment]
@magento import code to https://github.com/magento-engcom/magento2ce

**Comment by magento-engcom-team** [comment]
@gabrieldagama the pull request successfully imported.

**Comment by magento-engcom-team** [comment]
@gabrieldagama, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@gabrieldagama, an error occurred during the Pull Request import.

**Pull request merged**